benchmark/src/main/scala/keycloak/scenario/CommonSimulation.scala

```diff
@@ -27,6 +27,10 @@ abstract class CommonSimulation extends Simulation {
     var default = http
       .acceptHeader("application/json")
       .disableFollowRedirect
+      .shareConnections
+      //.connectionHeader("keep-alive")
+      .useAllLocalAddresses
```

Comment:
In getting this test up to 100k I ran into networking issues beyond around 30k.

This was due to running out of src ports on a single local IP in use.  

At different times I saw different things on netstat...

At one point, I saw an exhaustion of ports due to 30k TIME_WAIT sockets.

Then I enabled [sharedConnections](https://gatling.io/docs/gatling/reference/current/http/protocol/#shareconnections), and things seem to be improved by it only keeping around 4-5k connections open.

But then that quit being its behavior for some reason and I would see 10s of thousands of ESTABLISHED connections.  And that would happen with or without keep-alive.

I believe it's probably affected by how well the server can keep up.  If it can service the requests faster than gatling needs to create a new request, then (with shareConnections enabled) it can reuse a connection from the pool.  But if the server can't keep up, it's a cascading problem because gatling must make a new connection to the server, incurring some extra load.

So, further, I enabled useAllLocalAddresses so that I could define new srcIPs for it to have available so there can be more than the 30k (based on ephemeral port range) between a single local IP and the single server IP/port.
e.g. ```
            ip addr add 192.168.40.1/32 dev lo label lo:40
            ip addr add 192.168.41.1/32 dev lo label lo:41
            ip addr add 192.168.42.1/32 dev lo label lo:42
            ip addr add 192.168.43.1/32 dev lo label lo:43
            ip addr add 192.168.44.1/32 dev lo label lo:44
```

Also, on my system I set `sysctl net.ipv4.tcp_tw_reuse=1` so that it can make use of TIME_WAIT sockets if necessary.

Reply:
@ddurham2 - It looks to me as if you're hitting a limit of the local minikube setup. In the parent issue I see that you created 6 pods, and then ran a high load against that setup.

While the local minikube setup is IMHO great to develop scenarios, reliable performance numbers of large systems are only available in a production-like setup, which would require a cloud setup. 

Changing parameters like `shareConnections` might make the scenario pass on minikube in a local setup, but won't show the right load created due to TLS connection reuse. So I'd rather recommend to scale down the scenario when running it on minikube.

For a setup to run a production load against that number of pods, I'd recommend using a cloud setup. 
Unfortunately those test setups then require someone to pay the cloud bill. 

Recently this project added scripts to deploy small EC2 instances to drive the load, see [Running benchmarks via Ansible and EC2](https://www.keycloak.org/keycloak-benchmark/benchmark-guide/latest/run/running-benchmark-ansible). There are also instructions on how to deploy an [OpenShift instance with ROSA to run Keycloak](https://www.keycloak.org/keycloak-benchmark/kubernetes-guide/latest/prerequisite/prerequisite-rosa).

Reply:
@ddurham2 - one more thought about `shareConnections` - if you think this is essential for your testing, you could place it behind an option, document it, and keep the default as before.

Reply:
I agree with keeping the load parameters low enough to suit the local setup and one of the other future alternatives is to extend the `Ansible playbook` to work with regular on-prem nodes or the cloud provider of your choice., but that is something we wont be prioritizing soon enough.

Reply:
done